lib/database/exceptions.rb

```diff
@@ -159,6 +159,8 @@ class DatabaseLoadFailureException < RavenException
   end
   class DatabaseNotFoundException < RavenException 
   end
+  class NotSupportedException < RavenException
```

Comment:
What happens when you run into an exception that you aren't already familiar with?
Do we still get good error?

Reply:
@ayende NotSupportedException is just an "temporary" exception for situation when we're trying to connect to secured server (which is not supported yet). After i implement auth, it will be removed

Reply:
I'm really asking about the parallel exception hierarchy and what happens if an exception is sent from the server that you don't have a class for.